cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1269 @@
+	var resp []ListData
+	err := common.SendHTTPGetRequest(pathGetAllLists+idBoard+apiKey+apiToken, true, verbose, &resp)
+	if err != nil {
+		return resp, err
+	}
+	return resp, nil
+}
+
+// TrelloCreateNewCard creates a new card on the list specified on trello
+func TrelloCreateNewCard(fillData *CardFill) error {
+	params := url.Values{}
+	params.Set("idList", fillData.ListID)
+	if fillData.Name != "" {
+		params.Set("name", fillData.Name)
+	}
+	if fillData.Desc != "" {
+		params.Set("desc", fillData.Desc)
+	}
+	if fillData.Pos != "" {
+		params.Set("pos", fillData.Pos)
+	}
+	if fillData.Due != "" {
+		params.Set("due", fillData.Due)
+	}
+	if fillData.MembersID != "" {
+		params.Set("idMembers", fillData.MembersID)
+	}
+	if fillData.LabelsID != "" {
+		params.Set("idLabels", fillData.LabelsID)
+	}
+	_, err := common.SendHTTPRequest(http.MethodPost,
+		pathNewCard+params.Encode()+apiKey+apiToken,
+		nil,
+		nil)
+	return err
+}
+
+// TrelloCreateNewCheck creates a new checklist item within a given checklist from trello
+func TrelloCreateNewCheck(newCheck string) error {
+	params := url.Values{}
+	params.Set("name", newCheck)
+	_, err := common.SendHTTPRequest(http.MethodPost,
+		pathChecklists+trelloChecklistID+params.Encode()+apiKey+apiToken,
+		nil,
+		nil)
+	return err
+}
+
+// TrelloGetChecklistItems get info on all the items on a given checklist from trello
+func TrelloGetChecklistItems() (ChecklistItemData, error) {
+	var resp ChecklistItemData
+	path := fmt.Sprintf(pathChecklistItems, trelloChecklistID, configData.Key, configData.Token)
+	return resp, common.SendHTTPGetRequest(path, true, verbose, &resp)
+}
+
+// NameStateChanges returns the appropriate update name & state for trello (assumes single digit updates pending)
+func NameStateChanges(currentName, currentState string) (string, error) {
+	r, err := regexp.Compile(`[\s\S]* \d{1}$`) // nolint: gocritic
+	if err != nil {
+		return "", err
+	}
+	var tempNumber int64
+	var finalNumber string
+	if r.MatchString(currentName) {
+		stringNum := string(currentName[len(currentName)-1])
+		tempNumber, err = strconv.ParseInt(stringNum, 10, 64)
+		if err != nil {
+			return "", err
+		}
+		if tempNumber != 1 || currentState != complete {
+			tempNumber++
+		} else {
+			tempNumber = 1
+		}
+		finalNumber = strconv.FormatInt(tempNumber, 10)
+	}
+	byteNumber := []byte(finalNumber)
+	byteName := []byte(currentName)
+	byteName = byteName[:len(byteName)-1]
+	byteName = append(byteName, byteNumber[0])
+	return string(byteName), nil
+}
+
+// TrelloUpdateCheckItem updates a check item for trello
+func TrelloUpdateCheckItem(checkItemID, name, state string) error {
+	params := url.Values{}
+	newName, err := NameStateChanges(name, state)
+	if err != nil {
+		return err
+	}
+	params.Set("name", newName)
+	params.Set("state", incomplete)
+	path := fmt.Sprintf(pathUpdateItems, trelloCardID, checkItemID, params.Encode(), configData.Key, configData.Token)
```

Comment:
These vars won't be populated if a person sets these in the config but doesn't use the cmdline